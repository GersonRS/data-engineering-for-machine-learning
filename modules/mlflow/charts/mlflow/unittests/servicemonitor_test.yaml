suite: test servicemonitor

templates:
  - servicemonitor.yaml

tests:
  - it: should be empty when serviceMonitor is not enabled
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    asserts:
      - hasDocuments:
          count: 0

  - it: should be empty when capability has not required api version
    set:
      serviceMonitor.enabled: true
    asserts:
      - hasDocuments:
          count: 0

  - it: should use the desired service name as the target endpoint
    set:
      serviceMonitor.enabled: true
      service.name: myendpoint
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    asserts:
      - equal:
          path: spec.endpoints[0].targetPort
          value: myendpoint

  - it: should use a service endpoint rather than a pod endpoint
    set:
      serviceMonitor:
        enabled: true
        useServicePort: true
      service.name: anothermyendpoint
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    asserts:
      - equal:
          path: spec.endpoints[0].port
          value: anothermyendpoint

  - it: should match snapshot of default values
    set:
      serviceMonitor.enabled: true
    release:
      name: mlflow
      namespace: mlflow
    capabilities:
      apiVersions:
        - monitoring.coreos.com/v1
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
