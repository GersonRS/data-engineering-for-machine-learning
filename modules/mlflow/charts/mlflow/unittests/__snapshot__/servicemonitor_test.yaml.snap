should match snapshot of default values:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        release: prometheus
      name: mlflow
      namespace: monitoring
    spec:
      endpoints:
      - interval: 30s
        path: /metrics
        scrapeTimeout: 10s
        targetPort: http
      jobLabel: mlflow
      namespaceSelector:
        matchNames:
        - mlflow
      selector:
        matchLabels:
          app: mlflow
          app.kubernetes.io/instance: mlflow
          app.kubernetes.io/name: mlflow
